srvutil/metrics.go

```diff
@@ -56,41 +80,50 @@ func (w *hijackableRecorder) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	return w.httpRecorder.ResponseWriter.(http.Hijacker).Hijack()
 }
 
-func newHTTPRecorder(w http.ResponseWriter) loggableHTTPRecorder {
-	recorder := httpRecorder{ResponseWriter: w}
+func newHTTPRecorder(w http.ResponseWriter, bodyLogPredicate BodyLogPredicateFunc) loggableHTTPRecorder {
+	recorder := httpRecorder{ResponseWriter: w, bodyLogPredicate: bodyLogPredicate}
 	if _, ok := w.(http.Hijacker); ok {
 		return &hijackableRecorder{recorder}
 	}
 	return &recorder
 }
 
-// RequestMetricsMiddleware records the time taken to serve a request.
+// RequestMetricsMiddleware records the time taken to serve a request, and logs request and response data.
```

Comment:
Would it make sense to introduce another request/response logging middleware instead and keep this one as time metrics only?

Reply:
The problem is around how we need to wrap response writer recorder to get access to this data. We can wrap it multiple times, but then all wrappers need to take care to support the additional interfaces like `http.Hijacker`.